### PR TITLE
Development

### PR DESCRIPTION
**Pull request by htuerker**
enumerable_methods project's pull request is blow there.
[Enumerable Methods](https://github.com/btuerker/enumerable_methods/pull/2)

### CODE REVIEW

File: spec/lib/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,252 @@
+require_relative '../spec_helper'
+
+RSpec.describe Board do
+  describe "#initialize" do
+    it "should create two dimensional board_state array with given size parameter" do
+      board = Board.new(3)
```
Comment (heyapricot) [inline spec/lib/board_spec.rb]: [Let is your friend ](http://www.betterspecs.org/#let)

File: spec/lib/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      expect(board.valid_move?('13')).to be true
+      expect(board.valid_move?('26')).to be false
+    end
+  end
+
+  describe "#translate_move" do
+    it "should return translated move from number to coordinates" do
+      board = Board.new(3)
+      expect(board.translate_move(1)).to eql([0,0])
+      expect(board.translate_move(4)).to eql([1,0])
```
Comment (heyapricot) [inline spec/lib/board_spec.rb]: If you want to be super deterministic you could generate the mapping of the 9 squares and check all 9 of them. This is not necessary, just an idea

File: spec/lib/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+      board.make_move('21', GameConstants::PLAYER_ONE_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_ONE_WON)
+
+      board = Board.new(4)
+      board.make_move('1', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('5', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('9', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('13', GameConstants::PLAYER_TWO_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_TWO_WON)
+    end
```
Comment (heyapricot) [inline spec/lib/board_spec.rb:132]: Why not writing `board.make_move` inside a loop and generating the combinations dynamically. Rewriting this if needed would be painful

**Review by heyapricot — APPROVED** [review]
# Don't you love tests? 📝
Nice work with the tests. Hope you had fun writing them. I liked the approach. Left some minor comments for your consideration

File: spec/lib/board_spec.rb (hunk 3 above)
Comment (htuerker) [reply spec/lib/board_spec.rb:132]: we tried to change it with a loop but we both agreed to use basically make_move is better. 

` 3.times do
        board.make_move(counter.to_s, GameConstants::PLAYER_ONE_MARK)
        counter += 3
      end
`

File: spec/lib/board_spec.rb (hunk 2 above)
Comment (htuerker) [reply spec/lib/board_spec.rb]: We did it, I agreed with you to add full tests

File: spec/lib/board_spec.rb (hunk 1 above)
Comment (htuerker) [reply spec/lib/board_spec.rb]: It's a good idea, we changed it through the way but I think our case doesn't fit with it :) I know what you wanted to say.

**Pull request merged**